### PR TITLE
Event creation and listing on the home page

### PR DESCRIPTION
**Pull request by Jezerig**
Added:
- Backend for creating events
- MUI DateTimePicker for start time, end time and joining deadline
- Joining deadline is the last time a person can join/purchase a ticket for the event (has to be before or equal to the start time)
- Switch for setting the deadline (default is the same as the start time)
- End time is not required but can be set (has to be after the start time)
- Switch for limiting tickets and a inputfield to set the amount of tickets
- Price of the event
- Listing events in the home page

### CODE REVIEW

**Review by jannejjj — APPROVED** [review]
Looks very good to me! The nested `if`s when saving a new event aren't the most beautiful thing i've ever seen but they're not that hard to read (and i can't think of a clearer way to do it either 😄).

Just a slight nitpick, in the modal for creating a new event, the joining deadline input is overlapped by the ending time input when it's active:
image
 
 I'll approve anyway, just fix that before merging ✌

File: client/src/components/Home.js [hunk 1]
```diff
@@ -291,49 +311,173 @@ function Home(props) {
...
+  // Only for users that have logged in
+  if (props.currentUser.loggedIn)
+  {
+    fetchEvents();
+    return () => {
+        mounted = false;
+    };
+  }
+  setLoading(false);
+  }, [])
```
[the remark this answers is not in the document]
Comment (jannejjj) [reply client/src/components/Home.js:423]: Nevermind, it looks like this is a part of the currentUser issue I discovered and will be fixed with that.

[the remark this answers is not in the document]
Comment (jannejjj) [reply client/src/components/Home.js:423]: Well it's definitely good enough in my opinion, could not have done any better if I tried I'm sure 🤪

**Comment by Nesssu** [comment]
Once a new event has been created, it would be nice if the home page would be updated to view the new event automatically without the need to refresh the page manually. The easiest way to do this is to use the newEventModal boolean variable. But you can also create a new "update" variable if you think it's a more proper way to do it. Below is an example of how to do it with the newEventModal variable.

AssocEase_HomePageRefresh

This refreshes the page everytime the page is manually refreshed and also when the add new event modal is closed. Of course this means that the page is refreshed also when new events have not been created but I don't think that it is a problem. 😄

**Review by Nesssu — APPROVED** [review]

**Pull request merged**